src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp

```diff
@@ -419,12 +420,33 @@ namespace OpenMS
     //################## DEFAULTS #################
     initializeDefaultParameters_();
 
+    // store defaults in param_
+    defaultsToParam_();
+
     // load param file
     loadPreferences();
 
     // set current path
     current_path_ = param_.getValue(user_section + "default_path").toString();
 
+    // set plugin search path, create it if it does not already exist
+    String plugin_path = String(param_.getValue(user_section + "plugins_path").toString());
+    if (!File::exists(plugin_path))
+    {
+      QString dir = plugin_path.suffix('/').toQString();
+      QDir path = QDir(plugin_path.toQString());
+      path.cdUp();
+      printf("Dir: %s\n", dir.toStdString().c_str());
+      printf("Path: %s\n", path.absolutePath().toStdString().c_str());
+      if (!path.mkdir(dir))
+      {
+        OPENMS_LOG_WARN << "Unable to create plugin directory at " << path.absolutePath().toStdString() << "\n";
+        param_.setValue(user_section + "plugins_path", ".");
+        plugin_path = ".";
+      }
+    }
```

Comment:
this should not be in TOPPViewBase.
Please move it into TVToolDiscovery, as part of `setPluginPath`.
There is also `QDir::mkpath` which should shorten the code considerably.

Also, what happens if the path could not be created?!
Does it cause problems down the road which should be addressed here?